src/bin/uefi.rs

```diff
@@ -149,6 +153,25 @@ fn init_logger(st: &SystemTable<Boot>) -> (PhysAddr, FrameBufferInfo) {
         .expect_success("failed to locate gop");
     let gop = unsafe { &mut *gop.get() };
 
+    let mode = {
+        let modes = gop.modes().map(Completion::unwrap);
+        match (
+            CONFIG.desired_framebuffer_height,
+            CONFIG.desired_framebuffer_width,
+        ) {
+            (Some(height), Some(width)) => modes
+                .filter(|m| m.info().resolution() == (width, height))
+                .last(),
+            (Some(height), None) => modes.filter(|m| m.info().resolution().1 == height).last(),
+            (None, Some(width)) => modes.filter(|m| m.info().resolution().0 == width).last(),
+            _ => None,
+        }
+    };
+    if let Some(mode) = mode {
+        gop.set_mode(&mode)
+            .expect_success("failed to apply gop mode");
```

Comment:
Might be better to replace this panic message with `Failed to apply the desired display mode`.

Reply:
You're right, that's probably clearer for users of the crate. Fixed